src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -493,15 +500,14 @@ private void updatePrimaryKeys(TableId tableId, List<String> primaryKeys) throws
       return;
     }
     primaryKeyStore.put(tableId, primaryKeys);
-    context.putState(String.format("bigquery-%s-%s", tableId.getDataset(), tableId.getTable()),
+    context.putState(getTableStateKey(tableId),
                      Bytes.toBytes(GSON.toJson(new BigQueryTableState(primaryKeys))));
```

Comment:
when primary key is updated, we don't preserve the old sortkey in the table state ?

Reply:
Fixed